lib/hexpm_web/controllers/dashboard/password_controller.ex

```diff
@@ -13,8 +13,11 @@ defmodule HexpmWeb.Dashboard.PasswordController do
 
     case Users.update_password(user, params["user"], audit: audit_data(conn)) do
       {:ok, _user} ->
+        breached? = Hexpm.Accounts.Pwned.password_breached?(params["user"]["password"])
```

Comment:
> The API is not throttled at all, so that should not be an issue. I added timeouts to the hackney call so this check will not block for a long time in the controllers. if it does not respond in 500ms, we treat the password as secure and hopefully we catch them next time around.

I think this is a good start.

I'd like to propose a different approach, perhaps as another iteration on this feature: when we log in (or change password), we make async request to check the password so that we never block. And when that request finishes and the password is pwned, we'd mark the user as pwned and show the flash message on all pages, not just when they're logging in. And if the async request failed we can try again with a backoff etc. (We could extend that even further, if request failed we could mark the user in DB appropriately, so we could keep checking even if node is restarted.)

WDYT?

Reply:
My approach is philosophically lenient. I would prefer to show the message sparingly unless we decide we really want to enforce stronger passwords (as opposed to encouraging them.) 

I considered a nonblocking call, but I think it's hard to reconcile that with this permissive approach. But the time the call responded the user could have advanced to the next page. It would also require websockets (at least the way I am imagining it.)

We should decide as a group how permissive or strict we want to be, and I think the correct approach will be clear after that.

How does that sound?

Reply:
Yeah, I guess it might better to either be lenient or enforce strong password, not try to nag people into changing them. Definitely looking forward to hearing from others.